bcda/suppression/suppression_test.go

```diff
@@ -0,0 +1,323 @@
+
+	suppressions := []models.Suppression{}
+	db.Find(&suppressions, "file_id = ?", suppressionFile.ID)
+	assert.Equal(4, len(suppressions))
+	assert.Equal("1000087481", suppressions[0].HICN)
+	assert.Equal("1-800", suppressions[0].SourceCode)
+	assert.Equal("1000093939", suppressions[1].HICN)
+	assert.Equal("1-800", suppressions[1].SourceCode)
+	assert.Equal("1000079734", suppressions[2].HICN)
+	assert.Equal("", suppressions[2].SourceCode)
+	assert.Equal("1000050218", suppressions[3].HICN)
+	assert.Equal("1-800", suppressions[3].SourceCode)
+	assert.Nil(err)
+	err = deleteFilesByFileID(suppressionFile.ID, db)
+	assert.Nil(err)
+
+	// negative
+	metadata = suppressionFileMetadata{}
+	err = importSuppressionData(metadata)
+	assert.NotNil(err)
+	assert.Contains(err.Error(), "could not read file")
+
+	filepath := BASE_FILE_PATH + "suppressionfile_MissingData/T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000011"
+	metadata = suppressionFileMetadata{
+		timestamp:    time.Now(),
+		filePath:     filepath,
+		name:         "T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000011",
+		deliveryDate: time.Now(),
+	}
+	err = importSuppressionData(metadata)
+	assert.NotNil(err)
+	assert.Contains(err.Error(), "failed to parse the effective date '20191301' from file: "+filepath)
+
+	filepath = BASE_FILE_PATH + "suppressionfile_MissingData/T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000012"
+	metadata = suppressionFileMetadata{
+		timestamp:    time.Now(),
+		filePath:     filepath,
+		name:         "T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000012",
+		deliveryDate: time.Now(),
+	}
+	err = importSuppressionData(metadata)
+	assert.NotNil(err)
+	assert.Contains(err.Error(), "failed to parse the samhsa effective date '20191301' from file: "+filepath)
+
+	filepath = BASE_FILE_PATH + "suppressionfile_MissingData/T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000013"
+	metadata = suppressionFileMetadata{
+		timestamp:    time.Now(),
+		filePath:     filepath,
+		name:         "T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000013",
+		deliveryDate: time.Now(),
+	}
+	err = importSuppressionData(metadata)
+	assert.NotNil(err)
+	assert.Contains(err.Error(), "failed to parse beneficiary link key from file: "+filepath)
+}
+
+func (s *SuppressionTestSuite) TestValidate() {
+	assert := assert.New(s.T())
+
+	// positive
+	suppressionfilePath := BASE_FILE_PATH + "synthetic1800MedicareFiles/test/T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000009"
+	metadata := suppressionFileMetadata{timestamp: time.Now(), filePath: suppressionfilePath}
+	err := validate(metadata)
+	assert.Nil(err)
+
+	// bad file path
+	metadata.filePath = metadata.filePath + "/blah/"
+	err = validate(metadata)
+	assert.NotNil(err)
+	assert.Contains(err.Error(), "could not read file "+metadata.filePath)
+
+	// invalid file header
+	metadata.filePath = BASE_FILE_PATH + "suppressionfile_BadHeader/T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000009"
+	err = validate(metadata)
+	assert.NotNil(err)
+	assert.EqualError(err, "invalid file header for file: "+metadata.filePath)
```

Comment:
`assert.EqualError()` checks for the error being non-nil, so we can leave out `assert.NotNil()` before it. https://godoc.org/github.com/stretchr/testify/assert#EqualError